### PR TITLE
Add fly controller

### LINKED ISSUE #364: Fly controller
Add a FlyController that works as follows:

* WASD to strafe sideways and move forward/backward
* RF to go up and down
* Mouse movement to look around

### PR DESCRIPTION
**Pull request by almarklein**
Closes #364 

Screen Recording 2023-04-13 at 22 57 05

### CODE REVIEW

File: docs/reference.rst [hunk 1]
```diff
@@ -40,6 +40,7 @@ Currently this includes the following classes, which comprise the public API:
     ~pygfx.controllers.PanZoomController
     ~pygfx.controllers.OrbitController
     ~pygfx.controllers.TrackballController
+    ~pygfx.controllers.OrbitController
```
Comment (Korijn) [inline docs/reference.rst:43]: I think this is wrong

File: pygfx/controllers/_base.py [hunk 2]
```diff
@@ -372,9 +374,8 @@ def _handle_button_down(self, key, action_tuple, viewport):
         if action is None:
             action = self._create_action(key, action_tuple, 0, None, viewport.rect)
             action.snap_distance = 0.01
-            if mode == "push":
-                action.done = True
-        if mode in ("push", "peak"):
+        action.done = True if mode == "push" else False
```
Comment (Korijn) [inline pygfx/controllers/_base.py:377]:
```suggestion
        action.done = mode == "push"
```

:D

File: pygfx/controllers/_fly.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+from ._base import Controller
+
+
+class FlyController(Controller):
+    """A controller to fly around a scene as if it's a flight simulator.
+
+    Default controls:
+
+    * Left mouse button: rotate.
+    * "wasd": move forward, backwards, and to the sides.
```
Comment (Korijn) [inline pygfx/controllers/_fly.py:14]:
```suggestion
    * "wasd": move forward, backwards, and strafe to the sides.
```

File: pygfx/controllers/_fly.py [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+        if speed is None:
+            cam_state = camera.get_state()
+            approx_scene_size = 0.5 * (cam_state["width"] + cam_state["height"])
+            scene_fly_thru_time = 5  # seconds
+            speed = approx_scene_size / scene_fly_thru_time
+        self.speed = speed
+
+    @property
+    def speed(self):
+        """The (maxium) speed that the camera will move, in units per second.
```
Comment (Korijn) [inline pygfx/controllers/_fly.py:49]:
```suggestion
        """The (maximum) speed that the camera will move, in units per second.
```

File: docs/reference.rst (hunk 1 above)
Comment (almarklein) [inline docs/reference.rst:43]:
```suggestion
    ~pygfx.controllers.FlyController
```

**Comment by kushalkolar** [comment]
This looks epic!

**Review by Korijn — APPROVED** [review]

**Pull request merged**